### PR TITLE
Finish basic tearing on nonlinear systems

### PR DESCRIPTION
**Pull request by YingboMa**
Check the `test/tearing.jl` file for an example.

### CODE REVIEW

File: src/tearing.jl [hunk 1]
```diff
@@ -0,0 +1,377 @@
+################################################
+#
+# Functions to tear systems of equations
```
Comment (ChrisRackauckas) [inline src/tearing.jl:3]: We need to add an acknowledgment in the license file.

Comment (ChrisRackauckas) [reply src/tearing.jl:3]: I think? @ViralBShah it's from an MIT license.

Comment (ViralBShah) [reply src/tearing.jl:3]: Which project is it from. Yes, we should document that in our LICENSE.

Comment (ChrisRackauckas) [reply src/tearing.jl:3]: Modia

**Pull request merged**